service/deploy.py

```diff
@@ -339,75 +357,77 @@ def user_deploy_install(instance_ip, username, instance_id, install_action, inst
-                  % (limit_playbooks, runner._get_files(playbook_dir))
+                  % (limit_playbooks, playbook_runner._get_files(playbook_dir))
         raise AnsibleDeployException(msg)
-    runner.run()
-    return runner
+    playbook_runner.run()
+    return playbook_runner
 
 
-def _one_runner_one_playbook_execution(
-        playbook_dir, host_file, extra_vars, my_limit,
-        logger=None, limit_playbooks=None, **runner_opts):
-    from subspace.runner import Runner
-    runners = [Runner.factory(
-            host_file,
-            os.path.join(playbook_dir, playbook_path),
-            extra_vars=extra_vars,
-            limit_hosts=my_limit,
-            logger=logger,
-            limit_playbooks=limit_playbooks,
-            # Use atmosphere settings
-            group_vars_map={
-                filename: os.path.join(settings.ANSIBLE_GROUP_VARS_DIR,
-                                       filename)
-                for filename in os.listdir(settings.ANSIBLE_GROUP_VARS_DIR)
-            },
-            private_key_file=settings.ATMOSPHERE_PRIVATE_KEYFILE,
-            **runner_opts)
-        for playbook_path in os.listdir(playbook_dir)
-        if not limit_playbooks or playbook_path in limit_playbooks]
-    [runner.run() for runner in runners]
-    return runners
+def _select_playbooks_from_path(playbook_path, limit_playbooks=[]):
+    """
+    Input:
+      - Path of playbooks directory
+      - A list of playbooks to be run.
+    Output:
+      An ordered list of playbook files.
+    """
+    if not isinstance(playbook_path, basestring):
+        raise TypeError(
+            "Expected 'playbook_path' as string,"
+            " received %s" % type(playbook_path))
+    # Convert file path to list of playbooks:
+    if not os.path.exists(playbook_path):
+        raise ValueError("Could not find path: %s" % (playbook_path,))
+
+    if os.path.isdir(playbook_path):
+        playbook_list = _select_playbooks_from_dir(
+            playbook_path, limit_playbooks)
+    else:
+        playbook_list = [playbook_path]
+    return playbook_list
+
+
+def _select_playbooks_from_dir(playbook_dir, limit=[]):
+    """
+    Given a directory, return all `.yml` playbooks
+    If a limit is specified, only include playbooks found in the limit.
+    """
+    return [playbook_path for playbook_path in _list_yml_files_in_dir(playbook_dir)
+            if not limit or playbook_path.split('/')[-1] in limit]
+
+
+def _list_yml_files_in_dir(directory):
+    """
+    Walk the directory and retrieve each yml file.
+    """
+    files = []
+    directories = list(os.walk(directory))
+    directories.sort(cmp=operator.lt)
+    for d in directories:
+        a_dir = d[0]
+        files_in_dir = d[2]
+        files_in_dir.sort()
+        if os.path.isdir(a_dir):
+            for f in files_in_dir:
+                if os.path.splitext(f)[1] == ".yml":
+                    files.append(os.path.join(a_dir, f))
+    return files
```

Comment:
A lot of this code could be removed if we leveraged the fact that you can create playbooks which contain other playbooks. We shouldn't be sorting by name. Could create a playbook instance_deploy.yml that includes the playbooks it needs.

Reply:
Interesting idea @cdosborn - Let's get this in _first_ and then make that change separately with a corresponding PR in `atmosphere-ansible`?